test_programs/compile_success_empty/comptime_negative_zero/src/main.nr

```diff
@@ -0,0 +1,9 @@
+fn main() {
+    comptime {
+        assert_eq(-0, 0);
+        assert_eq(-1 + 1, 0);
+        assert_eq(-1 - -1, 0);
+        assert_eq(0 * -1, 0);
+        assert_eq(0 / -1, 0);
```

Comment:
These weren't failing before (the issue was in `impl Neg for SignedField`) but I figured they were good for completeness.